packages/yew/src/functional/hooks/use_ref.rs

```diff
@@ -137,3 +138,11 @@ where
 pub fn use_node_ref() -> NodeRef {
     (*use_state(NodeRef::default)).clone()
 }
+
+/// This hook is used for obtaining a [`HtmlRef`].
+/// The reference persists across renders, but a rerender is not triggered if the underlying ref
+/// changes.
+#[hook]
+pub fn use_html_ref<T: 'static + ErasedStorage>() -> HtmlRef<T> {
+    (*use_state(HtmlRef::default)).clone()
```

Comment:
Is there a better way to implement this than adding even more layers of `Rc`s and interior mutability. I believe right now there's 3 (4, I don't exactly remember) `Rc`s and 2 `RefCell`s used here

Reply:
The ones inside `HtmlRef` I think make sense to have, as it should be cheap to share. For the ones inside `use_state`, I'd point to #2618